endpointmanager/cmd/endpointpopulator/main.go

```diff
@@ -22,14 +22,18 @@ func failOnError(errString string, err error) {
 
 func main() {
 	var endpointsFile string
-	if len(os.Args) != 1 {
+	var source string
+
+	if len(os.Args) != 2 {
 		endpointsFile = os.Args[1]
```

Comment:
if `len(os.Args)` is 0, this fails.

Did you mean `if len(os.Args) == 2` instead?

Reply:
oh whoops, yeah i didn't think that through. it's now `if len(os.Args) == 3` since the 2nd argument is the file and the third argument is the source